CMakeLists.txt

```diff
@@ -163,12 +163,26 @@ cmake_path(RELATIVE_PATH libPath
 set(CMAKE_INSTALL_RPATH $ORIGIN $ORIGIN/${relativeRpath})
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH true)
 
+
+
+
 if ( T8CODE_USE_SYSTEM_SC )
     find_package( SC REQUIRED )
 else()
     set( SC_BUILD_EXAMPLES ${T8CODE_BUILD_TPL_EXAMPLES} )
     set( SC_BUILD_TESTING ${T8CODE_BUILD_TPL_TESTS} )
 
+    # When executing CMake tests for libsc, i.e. check_c_source_compiles/check_cxx_source_compiles and others
+    # we need to ignore the -Werror option, since these tests often produce warnings (see check_mpicommshared.cmake for example).
+    # If theses warnings are interpreted as errors, the test fails even though we want it to pass.
+    # This behaviour caused a serious bug and did not initialize our shared memory. See https://github.com/DLR-AMR/t8code/issues/1985.
+    set( T8CODE_OLD_CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS}" )
```

Comment:
```suggestion
```
this should not be necessary, because the else from line 171 opens its own scope